src/status_im/ui/screens/wallet/send/sheets.cljs

```diff
@@ -0,0 +1,77 @@
+            [status-im.ui.components.chat-icon.screen :as chat-icon]
+            [status-im.ui.components.list.views :as list]
+            [status-im.ui.screens.wallet.accounts.views :as wallet.accounts]
+            [status-im.utils.utils :as utils.utils]))
+
+(views/defview assets [address]
+  (views/letsubs [{:keys [tokens]} [:wallet/visible-assets-with-values address]
+                  currency [:wallet/currency]]
+    [list/flat-list
+     {:data      tokens
+      :key-fn    (comp str :symbol)
+      :render-fn (wallet.accounts/render-asset
+                  (:code currency)
+                  #(do
+                     (re-frame/dispatch [:bottom-sheet/hide-sheet])
+                     (re-frame/dispatch [:set-in [:wallet/prepare-transaction :symbol] (:symbol %)])))}]))
+
+(defn render-account [account]
+  [list-item/list-item
+   {:icon     [chat-icon/custom-icon-view-list (:name account) (:color account)]
+    :title    (:name account)
+    :on-press #(do
+                 (re-frame/dispatch [:bottom-sheet/hide-sheet])
+                 (re-frame/dispatch [:set-in [:wallet/prepare-transaction :from] account]))}])
+
+(views/defview accounts-list []
+  (views/letsubs [{:keys [accounts]} [:multiaccount]]
+    [list/flat-list {:data      accounts
+                     :key-fn    :address
+                     :render-fn render-account}]))
+
+(defn- request-camera-permissions []
+  (let [options {:handler :wallet.send/qr-scanner-result
+                 :cancel-handler :wallet.send/qr-scanner-cancel}]
+    (re-frame/dispatch
+     [:request-permissions
+      {:permissions [:camera]
+       :on-allowed
+       #(do
+          (re-frame/dispatch [:bottom-sheet/hide-sheet])
+          (re-frame/dispatch [:set-in [:wallet/prepare-transaction :modal-opened?] true])
+          (re-frame/dispatch [:navigate-to :qr-scanner options]))
+       :on-denied
+       #(utils.utils/set-timeout
+         (fn []
+           (utils.utils/show-popup (i18n/label :t/error)
+                                   (i18n/label :t/camera-access-error)))
+         50)}])))
+
+(defn choose-recipient []
+  [react/view
+   (for [item [{:title    (i18n/label :t/accounts)
+                :icon     :main-icons/profile
+                :theme :action
+                :on-press #(do
+                             (re-frame/dispatch [:bottom-sheet/hide-sheet])
+                             (re-frame/dispatch [:set-in [:wallet/prepare-transaction :modal-opened?] true])
+                             (re-frame/dispatch [:navigate-to :select-account]))}
+               {:title    (i18n/label :t/scan-qr)
+                :icon     :main-icons/qr
+                :theme :action
+                :on-press request-camera-permissions}
+               {:title    (i18n/label :t/recipient-code)
+                :icon     :main-icons/address
+                :theme :action
+                :on-press #(do
+                             (re-frame/dispatch [:bottom-sheet/hide-sheet])
```

Comment:
is code that got moved? could you create an event to avoid the triple dispatch?

Reply:
will fix to one event